fox_scraper/fox_scraper.py

```diff
@@ -14,97 +14,75 @@
         :return: a list of items (Item class) of the entire catalog
         """
-        soup_main_categories = self.scrap_fox_main_categories()
+        main_categories_links = self.get_main_categories()
         catalog = []
-        soup_sub_categories = []
-        main_cat_num = 0
-        for soup_main_category in soup_main_categories:
-            soup_sub_categories += list(self.scrap_fox_sub_categories(soup_main_category, main_cat_num))
-            catalog.append(self.scrap_fox_catalog(soup_sub_categories, main_cat_num))
-            main_cat_num += 1
+        for main_cat_num in range(len(main_categories_links)):
+            sub_categories = self.get_sub_categories(main_categories_links[main_cat_num], main_cat_num)
+            category_catalog = self.get_category_catalog(sub_categories, main_cat_num)
+            catalog.append(category_catalog)
         return catalog
 
-    def scrap_fox_main_categories(self):
+    def get_main_categories(self):
         """
         this function will scrap all main catalog categories
         :return: a list of beutifulSoup objects
         """
-        soup_main_category = self.get_soup_object(self.get_web_content(self.site_address))
-        main_categories = []
-        for link in soup_main_category.find_all(class_='link padding_hf_v '):
-            main_categories.append(link.get('href'))
+        main_categories_content = self.get_parsed_web_content(self.site_address)
+        main_categories_links = []
+        for link in main_categories_content.find_all(class_='link padding_hf_v '):
+            main_categories_links.append(link.get('href'))
             self.main_categories.append(link.get_text().strip())
-        return main_categories
+        return main_categories_links
 
-    def scrap_fox_sub_categories(self, soup_category, category_num):
+    def get_sub_categories(self, category_link, category_num):
         """
-        this function will scrap all catalog per sub category
-        :param soup_category: sub-category link from the site
+        this function will scrap sub-catalog links per category
+        :param category_link: sub-category link from the site
         :param category_num: category number for item description
-        :return: a list of catalog items per sub-category
+        :return: a list of sub-categories link per main-category
         """
-        sub_categories = self.get_soup_object(self.get_web_content(str(soup_category)))
-        sub_catalog = []
-        self.sub_categories.append(category_num)
-        self.sub_categories[category_num] = []
-        for sub_category in sub_categories.find_all('option'):
-            self.sub_categories[category_num].append(self.verify_encoding((sub_category.get_text()).strip()))
-            sub_catalog.append(format(sub_category['value']))
-        return sub_catalog
+        sub_categories_content = self.get_parsed_web_content(str(category_link))
```

Comment:
I don't - I used it when I though there was a different between the string and the unicode
(but it works on both)
updated.